pkg/rbac/sharedData.go

```diff
@@ -210,3 +221,133 @@ func (shared *SharedData) GetManagedClusters(cache *Cache, ctx context.Context)
 	return shared.mcErr
 
 }
+
+func (cache *Cache) GetDisabledClusters(ctx context.Context) (*map[string]struct{}, error) {
+	userData, userDataErr := cache.GetUserData(ctx)
+	if userDataErr != nil {
+		return nil, userDataErr
+	}
+
+	if cache.SharedCacheDisabledClustersValid() {
+		klog.V(5).Info("Search Addon DisabledClusters Cache valid")
+	} else {
+		klog.V(5).Info("DisabledClusters Cache not valid") // - running query to get search addon disabled clusters")
+		//run query and get disabled clusters
+		if disabledClustersFromQuery, err := cache.findSrchAddonDisabledClusters(ctx); err != nil {
+			klog.Error("Error retrieving Search Addon disabled clusters: ", err)
+			cache.setDisabledClusters(map[string]struct{}{}, err)
+			return nil, err
+		} else {
+			cache.setDisabledClusters(*disabledClustersFromQuery, nil)
+		}
+	}
+
+	//check if user has access to disabled clusters
+	if userHasAccessToDisabledClusters(&cache.shared.disabledClusters, userData.ManagedClusters) {
+		klog.V(5).Info("user has access to Search Addon disabled clusters ")
+		cache.shared.dcLock.Lock()
+		defer cache.shared.dcLock.Unlock()
+		return &cache.shared.disabledClusters, cache.shared.dcErr
```

Comment:
When the user has access to 1 disabled cluster, we will return the full list of disabled clusters, which may include clusters not visible to the user.
Need to either:
- Return the correct list OR
- Update the function to return a boolean instead. (This might be enough for the messages scenario.)